smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -160,15 +165,17 @@ pub fn handle_withdraw_position_reply(
 ) -> Result<Response, ContractError> {
     let msg: MsgWithdrawPositionResponse = data.try_into()?;
 
-    // let msg: MsgWithdrawPositionResponse = data.into_result().unwrap().data.unwrap().try_into()?;
+    // TODO: First we should subtract strategist rewrds and send them out. then follow
```

Comment:
This didn't break the contract balance but instead increased share price of users, If we prepend correctly to other messages in the match of the contracts ExecuteMsg, this should not be an issue

Reply:
This has been solved by your prepend_claim patch and this TODO comment is not valid anymore. Deleting!